### PR TITLE
[ADD] accounting: add Hong Kong localisations

### PR DESCRIPTION
**Pull request by tong-odoo**
tast-3042940

### CODE REVIEW

**Comment by Donapi** [comment]
Hi @tong-odoo we will review asap and keep you informed.

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+=========
+
+Configuration
+=============
+
+:ref:`Install <general/install>` the :guilabel:`Hong Kong - Accounting` module to
+get the features of the HK localization.
+
+.. image:: hong_kong/hk_01.png
+   :align: center
```
Comment (Donapi) [inline content/applications/finance/fiscal_localizations/hong_kong.rst:12]:
```suggestion
.. image:: hong_kong/hk-module.png
   :align: center
   :alt: "Hong Kong - Accounting" localization module marked as "installed"
```

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+and activate the **QR Code** feature.
+
+.. image:: hong_kong/hk_02.png
+   :align: center
+
+Configure your FPS bank account
+-------------------------------
+
+Go to :menuselection:`Contacts --> Configuration --> Bank Accounts`,
+select a Account Holder with Hong Kong as country, open your FPS bank
```
Comment (Donapi) [inline content/applications/finance/fiscal_localizations/hong_kong.rst:30]:
```suggestion
select an account holder with Hong Kong as country, open your FPS bank
```
I no longer see where to "open your FPS bank account and fill in the "**FPS Type** and **FPS ID/Phone Number/Email Address** fields. Is there any configuration needed to be able to see it? It worked on the runbot you sent but not when I try to reproduce it in another runbot.

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Configure your Bank Account's journal
+-------------------------------------
+
+Go to :menuselection:`Accounting --> Configuration --> Journals`, open your
+bank journal, then fill out the Account Number and Bank under the Journal
+Entries tab.
+
+.. image:: hong_kong/hk_04.png
+   :align: center
```
Comment (Donapi) [inline content/applications/finance/fiscal_localizations/hong_kong.rst:45]: Suggestion to only add screenshots of the relevant area, rather than keeping wide areas with arrows or squares to show what we want to illustrate.

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+Issue Invoices with FPS QR Codes
+--------------------------------
+
+To enable the FPS QR Code on your invoices, go to :menuselection:`Accounting
+--> Customers --> Invoices`, and create a new invoice.
+
+Open the *Other Info* tab and select the *Payment QR-code* option to
```
[the remark this answers is not in the document]
Comment (Donapi) [reply content/applications/finance/fiscal_localizations/hong_kong.rst:53]: Use the guilabel markup to identify any text of the interactive user interface (e.g., button labels, view titles, field names, lists items, …).

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+*FPS Transfer QR*.
+
+.. image:: hong_kong/hk_05.png
+   :align: center
+
+Also check the *Recipient Bank* field and make sure that the account indicated
+is the one you want to use to receive your customer's payment.
+
+.. image:: hong_kong/hk_06.png
+   :align: center
```
Comment (Donapi) [inline content/applications/finance/fiscal_localizations/hong_kong.rst:63]: I wouldn't add this image as such. If you want to illustrate how the QR code displays on the invoice, I would  add a short explanation and then the image.

**Review by Donapi — CHANGES_REQUESTED** [review]
Hello Tong,

First of all, thank you for creating this documentation page, good job!
Here are some general comments that need to be taken into account when writing the doc:

- we write up to 100 characters per line (https://www.odoo.com/documentation/14.0/contributing/documentation/rst_guidelines.html?highlight=100%20characters#start-a-new-line-before-the-100th-character)
- add :alt: on images (https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html?highlight=alt#alt-tags)
-  compress your images using pngquant (https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html?highlight=alt#alt-tags)
-  rename image files (https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html?highlight=alt#alt-tags)
- add links and references to internal documentation 
- use guilabels when needed (Use the guilabel markup to identify any text of the interactive user interface (e.g., button labels, view titles, field names, lists items, …).

I sometimes made specific comments in my review but didn't comment each time an issue was recurrent. 
If you are not used to write documentation and feel like it will take you too much time, I am happy to implement the changes and push the PR. Just let me know :)

File: content/applications/finance/fiscal_localizations/hong_kong.rst (hunk 2 above)
Comment (tong-odoo) [reply content/applications/finance/fiscal_localizations/hong_kong.rst:30]: The feature is not yet deployed to master therefore is not existed yet on other Runbot.

**Comment by tong-odoo** [comment]
@Donapi Thanks for the review! I have updated the comments above. Please take a look. Since this QR code is a new feature, please feel free to let me know if you need the new Runbot to test out the features.

**Comment by xpl-odoo** [comment]
Hi @tong-odoo ! Let me know if you would need a review while @Donapi is off.

**Comment by tong-odoo** [comment]
@xpl-odoo Yes thanks. I updated the comments above.

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+Open the :guilabel:`Other Info` tab and select the  :guilabel:`Payment QR-code` option to *EMV
+Merchant-Presented QR-code*.
+
+.. image:: hong_kong/hk-qr-code-invoice-setting.png
+   :align: center
+   :alt: Select EMV Merchant-Presented QR-code option
+
+Make sure that the :guilabel:`Recipient Bank` is the one you want to use to receive your customer's
+payment as Odoo uses this field to generate the FPS QR Code.
```
Comment (xpl-odoo) [inline content/applications/finance/fiscal_localizations/hong_kong.rst:70]:
```suggestion
Ensure that the :guilabel:`Recipient Bank` is the one you configured, as Odoo uses this field to generate the FPS QR code.
```
I would rephrase to clarify that you must select the one you just configured, not something else. If you need to specify "the one you want to use to receive customers' payments", move that info to the "FPS bank account configuration" heading.

**Review by xpl-odoo — CHANGES_REQUESTED** [review]
Nice job @tong-odoo! A few comments to address and we are good to go ;)

File: content/applications/finance/fiscal_localizations/hong_kong.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+.. image:: hong_kong/hk-qr-code-setting.png
+   :align: center
+   :alt: "QR Codes" feature activated
+
+FPS bank account configuration
+------------------------------
+
+Go to :menuselection:`Contacts --> Configuration --> Bank Accounts`, select an account Holder with
+Hong Kong as country, open your FPS bank account and fill in the :guilabel:`FPS Type` and
+:guilabel:`FPS ID/Phone Number/Email Address` fields.
```
[the remark this answers is not in the document]
Comment (tong-odoo) [reply content/applications/finance/fiscal_localizations/hong_kong.rst:37]: Yes the module is not on 16.0 runbot yet. If you need to test it, I will rebuild a runbot on my branch for your testing. Please feel free to let me know if you need it.

**Comment by tong-odoo** [comment]
Hi @xpl-odoo, I updated all the issues based on your comments. Please take a look.
As mentioned above, please let me know if you want to test the feature on Runbot. I will rebuild my branch on Runbot for you to have a test.

**Review by xpl-odoo — APPROVED** [review]

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Comment by tong-odoo** [comment]
@AntoineVDV It seems this merged suddenly, however the old PR is not ready. Could I target master instead?

**Comment by AntoineVDV** [comment]
@tong-odoo It merged because the blocking PR on odoo/odoo was closed. If this PR should not have been merged, you can create another PR with a revert commit, and continue the work later in another branch.

**Comment by tong-odoo** [comment]
> @tong-odoo It merged because the blocking PR on odoo/odoo was closed. If this PR should not have been merged, you can create another PR with a revert commit, and continue the work later in another branch.

Thanks, i will create a revert pr tonight

**Comment by tong-odoo** [comment]
@AntoineVDV I created a new revert PR #4304. It should target 16.0 only since only 16.0 is pushed the new commit. Please take a look. Thanks!
